src/fromager/bootstrapper.py

```diff
@@ -200,7 +201,14 @@ def bootstrap(self, req: Requirement, req_type: RequirementType) -> Version:
                     version=resolved_version,
                 )
             else:
+                logger.debug(f"have cached wheel in {unpacked_cached_wheel}")
                 sdist_root_dir = unpacked_cached_wheel
+                if sdist_root_dir.parent == self.ctx.work_dir:
+                    # TODO: investigate why this is happening for bootstrapping from tag
+                    logger.warning(
+                        f"{sdist_root_dir} should be two levels away from {self.ctx.work_dir}"
+                    )
+                    sdist_root_dir = sdist_root_dir / f"{req.name}-{resolved_version}"
```

Comment:
@dhellmann This workaround is required for `test_bootstrap_build_tags.sh`. Otherwise it creates places stuff like the Python build env in `ctx.work_dir`. I don't have the capacity to investigate.